drivers/spi/spi_ll_stm32.c

```diff
@@ -835,6 +837,17 @@ static int spi_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	if (spi == SPI1 || spi == SPI2 || spi == SPI3) {
+		/* SPI 1, 2, and 3 defaults clock source to PLL1
+		 * where the drivers expects the PCLK. Other sources
+		 * than the PCLK cannot be used as they are not defined
+		 * in `stm32_clock.h`, and thus not selectable in the DTS.
```

Comment:
That's a bit more complex: It's not only a `stm32_clock.h` issue.
What is present in dts is the clock gating (cf $9.5.11: Peripheral Clock Gating Control). To say it quick: On H7 some peripheral requires 2 clocks to be configured. This could not be done in dts yet, so the second clock configuration currently relies on SoC reset default value.